### PR TITLE
CCK: Reconcile rules

### PR DESCRIPTION
**Pull request by luke-hill**
`rules` is now is consistent for ruby/js.

I also massively removed the contradictory steps. Now it's at least consistent.

### ⚡️ What's your motivation? 

Goes towards completion of issue in tracker

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)
- :bug: Bug fix (non-breaking change which fixes a defect)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by luke-hill** [comment]
@davidjgoss think my knowledge of the `assert` library might be a bit lacking here. This will probably need a fix. I couldn't find a `beEmpty()` function

File: devkit/samples/rules/rules.feature.ts [hunk 1]
```diff
@@ -1,33 +1,28 @@
...
+  this.stock = []
 })
 
-When('the customer tries to buy a {float} chocolate with a {float} coin', function (price, paid) {
-  // TODO: implement this
-  assert(price)
-  assert(paid)
+When('the customer tries to buy a {int} cent chocolate bar', function (price) {
+  if(this.money > price) {
+    this.chocolate = this.stock.pop
```
Comment (davidjgoss) [inline devkit/samples/rules/rules.feature.ts:18]: This is referencing a method without calling it. I think it needs to be:

```ts
this.chocolate = this.stock.pop()
```

File: devkit/samples/rules/rules.feature.ts [hunk 2]
```diff
@@ -1,33 +1,28 @@
...
-  // TODO: implement this
+  assert(this.chocolate).toBe(null)
 })
 
-Then("the customer's change should be {int} {float} coin(s)", function (count, denomination) {
-  // TODO: implement this
-  assert(count)
-  assert(denomination)
+Then('the sale should happen', function () {
+  assert(this.chocolate).notToBe(null)
```
Comment (davidjgoss) [inline devkit/samples/rules/rules.feature.ts:27]:
```ts
assert.ok(this.chocolate)
```

File: devkit/samples/rules/rules.feature.ts [hunk 3]
```diff
@@ -1,33 +1,28 @@
...
-  assert(paid)
+When('the customer tries to buy a {int} cent chocolate bar', function (price) {
+  if(this.money > price) {
+    this.chocolate = this.stock.pop
+  }
 })
 
 Then('the sale should not happen', function () {
-  // TODO: implement this
+  assert(this.chocolate).toBe(null)
```
Comment (davidjgoss) [inline devkit/samples/rules/rules.feature.ts:23]:
```ts
assert.strictEqual(this.chocolate, undefined)
```

File: devkit/samples/rules/rules.feature.ts [hunk 4]
```diff
@@ -1,33 +1,28 @@
...
+Given('there are no chocolate bars in stock', function () {
+  this.stock = []
 })
 
-When('the customer tries to buy a {float} chocolate with a {float} coin', function (price, paid) {
-  // TODO: implement this
-  assert(price)
-  assert(paid)
+When('the customer tries to buy a {int} cent chocolate bar', function (price) {
+  if(this.money > price) {
```
Comment (davidjgoss) [inline devkit/samples/rules/rules.feature.ts:17]: I guess the check should really be `this.money >= price`?

**Review by davidjgoss — CHANGES_REQUESTED** [review]

**Review by davidjgoss — APPROVED** [review]

**Pull request merged**